subgraphs/_reference_/src/sdk/lending/manager.ts

```diff
@@ -0,0 +1,1065 @@
+    transfer.nonce = this.event.transaction.nonce;
+    transfer.logIndex = this.event.logIndex.toI32();
+    transfer.gasPrice = this.event.transaction.gasPrice;
+    transfer.gasUsed = this.event.receipt ? this.event.receipt!.gasUsed : null;
+    transfer.gasLimit = this.event.transaction.gasLimit;
+    transfer.blockNumber = this.event.block.number;
+    transfer.timestamp = this.event.block.timestamp;
+    transfer.sender = sender;
+    transfer.receiver = receiver;
+    transfer.market = this.market.id;
+    transfer.positions = [receiverPosition.getPositionID()!, positionID!];
+    transfer.asset = asset;
+    transfer.amount = amount;
+    transfer.amountUSD = amountUSD;
+    transfer.save();
+
+    this.snapshots.updateTransactionData(
+      TransactionType.TRANSFER,
+      amount,
+      amountUSD
+    );
+    this.snapshots.updateUsageData(TransactionType.TRANSFER, sender);
+
+    return transfer;
+  }
+
+  createFlashloan(
+    asset: Address,
+    account: Address,
+    amount: BigInt,
+    amountUSD: BigDecimal
+  ): Flashloan {
+    const flashloaner = new AccountManager(account);
+    if (flashloaner.isNewUser()) {
+      this.protocol.cumulativeUniqueUsers += INT_ONE;
+      this.protocol.save();
+    }
+    flashloaner.countFlashloan();
+
+    const flashloan = new Flashloan(
+      this.event.transaction.hash
+        .concatI32(this.event.logIndex.toI32())
+        .concatI32(Transaction.FLASHLOAN)
+    );
+    flashloan.hash = this.event.transaction.hash;
+    flashloan.nonce = this.event.transaction.nonce;
+    flashloan.logIndex = this.event.logIndex.toI32();
+    flashloan.gasPrice = this.event.transaction.gasPrice;
+    flashloan.gasUsed = this.event.receipt ? this.event.receipt!.gasUsed : null;
+    flashloan.gasLimit = this.event.transaction.gasLimit;
+    flashloan.blockNumber = this.event.block.number;
+    flashloan.timestamp = this.event.block.timestamp;
+    flashloan.account = account;
+    flashloan.market = this.market.id;
+    flashloan.asset = asset;
+    flashloan.amount = amount;
+    flashloan.amountUSD = amountUSD;
+    flashloan.save();
+
+    this.snapshots.updateTransactionData(
+      TransactionType.FLASHLOAN,
+      amount,
+      amountUSD
+    );
+    this.snapshots.updateUsageData(TransactionType.FLASHLOAN, account);
+
+    return flashloan;
+  }
+
+  //////////////////
+  //// Updaters ////
+  //////////////////
+
+  updateRewards(rewardData: RewardData): void {
+    if (!this.market.rewardTokens) {
+      this.market.rewardTokens = [];
+      this.market.rewardTokenEmissionsAmount = [];
+      this.market.rewardTokenEmissionsUSD = [];
+    }
+
+    // update market reward tokens with rewardData so that it is in alphabetical order
+    let rewardTokens = this.market.rewardTokens!;
+    let rewardTokenEmissionsAmount = this.market.rewardTokenEmissionsAmount!;
+    let rewardTokenEmissionsUSD = this.market.rewardTokenEmissionsUSD!;
+
+    for (let i = 0; i < rewardTokens.length; i++) {
```

Comment:
If you compare C with A `index` would be > 0. Then it would get to the last `else` and `0 != 2` so it would not break the for loop.

The case that it doesn't break the for loop is if the insert value is greater than the i-th index and is not at the end of the the `rewardTokens` list yet

Reply:
Ah sorry, I missed the `if` inside the last `else`. It makes sense then